esp-wifi/src/wifi/os_adapter.rs

```diff
@@ -33,42 +35,58 @@ use crate::compat::common::syslog;
 
 use super::WifiEvent;
 
-pub(crate) static mut WIFI_STATE: i32 = -1;
+pub(crate) static STA_STATE: AtomicWifiState = AtomicWifiState::new(WifiState::Invalid);
+pub(crate) static AP_STATE: AtomicWifiState = AtomicWifiState::new(WifiState::Invalid);
 
 // useful for waiting for events - clear and wait for the event bit to be set again
 pub(crate) static WIFI_EVENTS: Mutex<RefCell<EnumSet<WifiEvent>>> =
     Mutex::new(RefCell::new(enumset::enum_set!()));
 
 pub fn is_connected() -> bool {
-    unsafe { WIFI_STATE == wifi_event_t_WIFI_EVENT_STA_CONNECTED as i32 }
+    unsafe { get_sta_state() == WifiState::StaConnected }
 }
 
-#[derive(Debug, Clone, Copy)]
+#[atomic_enum]
```

Comment:
`atomic_enum` uses `core::sync::atomic` which is in turn implemented by emulation, sometimes. Since we're load/storing only here, maybe that's okay.

I have made a `portable_atomic_enum` crate, that is, ironically, incompatible with emulated atomic ops. Today was a good day 👍